### PR TITLE
[AzureMonitorDistro] Rename Azure.Monitor.OpenTelemetry to Azure.Monitor.OpenTelemetry.AspNetCore

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Collaborated with .NET and Azure SDK teams. Received recommendation to rename `Azure.Monitor.OpenTelemetry` to `Azure.Monitor.OpenTelemetry.AspNetCore`.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo/Program.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /*
 builder.Services.AddAzureMonitor(o =>
 {
-    o.ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-000000000000";
+    o.ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-00000000CODE";
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo/Program.cs:18]: any particular reason to change this?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo/Program.cs:18]: I see the change below.

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo/Program.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo/Program.cs:18]: makes it easier to understand, as we have entry in both file and appSettings. Personally, felt 123 is not a good.

**Review by cijothomas — COMMENTED** [review]
LGTM

**Comment by TimothyMothra** [comment]
TODO: need to review the CodeOwners file

**Pull request merged**